sklearn/tree/tree.py

```diff
@@ -108,6 +108,14 @@ def __init__(self,
         self.class_weight = class_weight
         self.presort = presort
 
+    def get_max_depth(self):
+        check_is_fitted(self, 'tree_')
+        return self.tree_.max_depth
```

Comment:
Isn't this the constructor argument, not the actual depth?

Reply:
That's what I thought, but it's set by the estimators as the `max_seen_depth` increases, and according to my few tests, it's actually the `max_depth` of the tree.

Reply:
I agree it's not the best name for the attribute though. I'd rather have it as `max_seen_depth` or something, maybe.

Reply:
how about we just call it ``depth`` and the getter ``get_depth``? Because it's the depth of the tree...

The ``tree_.max_depth`` argument is really terribly named (I didn't see the assignment in the builder). I would argue we should rename it, but maybe it's not worth it?

At least we should change the docstring to ``depth of the tree``.

Reply:
I updated the docstring. I can rename it if you want, but I think the clarification should do, and from the perspective of a tree (and not a decision tree), max_depth is explicit about the fact that it's not a balanced tree, which I kinda like.

Reply:
I would say keep the name in ``Tree`` but change it here. I don't think ``max_depth`` is necessarily explicit about that. And I don't see why someone would assume the tree is balanced. The depth of a tree has a very clear definition, and it's exactly this.